cpp/dolfinx/refinement/interval.h

```diff
@@ -0,0 +1,235 @@
+
+    if (refinement_marker[edge])
+    {
+      // find (global) index of new midpoint vertex:
+      // a --- c --- b
+      auto it = new_vertex_map.find(edge);
+      assert(it != new_vertex_map.end());
+      const std::int64_t c = it->second;
+
+      // add new edges to refined topology
+      edge_topology.insert(edge_topology.end(), {a, c, c, b});
+      parent_edge.insert(parent_edge.end(), {edge, edge});
+    }
+    else
+    {
+      // copy the previous edge
+      edge_topology.insert(edge_topology.end(), {a, b});
+      parent_edge.push_back(edge);
+    }
+  }
+
+  assert(edge_topology.size() == refined_cell_count * 2);
+  assert(parent_edge.size() == refined_cell_count);
+
+  std::vector<std::int32_t> offsets(refined_cell_count + 1);
+  std::ranges::generate(offsets, [i = 0]() mutable { return 2 * i++; });
+
+  graph::AdjacencyList cell_adj(std::move(edge_topology), std::move(offsets));
+
+  return {std::move(cell_adj), std::move(new_vertex_coords), xshape,
+          std::move(parent_edge)};
+}
+
+} // namespace impl
+
+/// Refines a (topologically) one dimensional mesh by splitting edges.
+///
+/// @param[in] mesh Mesh to be refined
+/// @param[in] edges Optional indices of the edges that should be split by this
+/// refinement. If not provided, all edges are considered marked for refinement,
+/// i.e. a uniform refinement is performed.
+/// @param[in] redistribute Option to enable redistribution of the refined mesh
+/// across processes.
+///
+/// @return Refined mesh, and list of parent edges - for every new edge index
+/// this contains the associated edge index of the pre-refinement mesh.
+template <std::floating_point T>
+std::tuple<mesh::Mesh<T>, std::vector<std::int32_t>>
+refine_interval(const mesh::Mesh<T>& mesh,
+                std::optional<std::span<const std::int32_t>> edges,
+                bool redistribute)
+{
+
+  if (mesh.topology()->cell_type() != mesh::CellType::interval)
+    throw std::runtime_error("Cell type not supported");
+
+  if (!mesh.topology()->index_map(1))
+    throw std::runtime_error("Edges must be initialised");
+
+  assert(mesh.topology()->dim() == 1);
+
+  auto [cell_adj, new_coords, xshape, parent_cell]
+      = impl::compute_interval_refinement(mesh, edges);
+
+  if (dolfinx::MPI::size(mesh.comm()) == 1)
+  {
+    return {mesh::create_mesh(mesh.comm(), cell_adj.array(),
+                              mesh.geometry().cmap(), new_coords, xshape,
+                              mesh::GhostMode::none),
+            std::move(parent_cell)};
+  }
+  else
+  {
+    // Check if mesh has ghost cells on any rank
+    // FIXME: this is not a robust test. Should be user option.
+    const int num_ghost_cells = mesh.topology()->index_map(1)->num_ghosts();
+    int max_ghost_cells = 0;
+    MPI_Allreduce(&num_ghost_cells, &max_ghost_cells, 1, MPI_INT, MPI_MAX,
+                  mesh.comm());
+
+    // Build mesh
+    const auto ghost_mode = max_ghost_cells == 0
```

Comment:
This is copying an existing design flaw. I think it would be better to make `ghost_mode` a function parameter, possibly with a default value of none.